### PR TITLE
Validate token applying to TomoZ/TomoX

### LINKED ISSUE #235: Node get stuck if read token decimals fails
Need to fixs:
- [ ] node need to keep working if read token decimals fails 
- [ ] validate decimals abi for `apply` tomox, tomox tx.

Logs:
```
2|node01  | DEBUG[04-14|09:25:17] ProcessLiquidationData time              tradeIds=0
2|node01  | DEBUG[04-14|09:25:17] getMediumTradePriceBeforeEpoch           baseToken=0x260eD86C808f0de23721Abf47039C8dA53C0be2f quoteToken=0x0000000000000000000000000000000000000001 inversePrice=0
2|node01  | DEBUG[04-14|09:25:17] Getting lendTokenTOMOPrice from tomox    lendTokenTOMOPrice=<nil> err=nil
2|node01  | DEBUG[04-14|09:25:17] getMediumTradePriceBeforeEpoch           baseToken=0xb3068564B614eA680A80F8d0089727949eA12B8B quoteToken=0x260eD86C808f0de23721Abf47039C8dA53C0be2f inversePrice=0
2|node01  | DEBUG[04-14|09:25:17] getMediumTradePriceBeforeEpoch           baseToken=0xb3068564B614eA680A80F8d0089727949eA12B8B quoteToken=0x0000000000000000000000000000000000000001 inversePrice=0
2|node01  | DEBUG[04-14|09:25:17] Getting collateralTOMOPrice from tomox   collateralTOMOPrice=<nil> err=nil
2|node01  | DEBUG[04-14|09:25:17] VM returned with error                   action="contract call" contract address=0x260eD86C808f0de23721Abf47039C8dA53C0be2f gas=977980   gasPrice=0 nonce=1 err="evm: execution reverted"
2|node01  | DEBUG[04-14|09:25:17] GetTokenDecimal from                     relayerSMC=0x33aEaC1209a4dD857833026D2f60D149E740FC7d tokenAddr=0x260eD86C808f0de23721Abf47039C8dA53C0be2f decimals=0
2|node01  | DEBUG[04-14|09:25:17] GetTokenDecimal                          lendingToken=0x260eD86C808f0de23721Abf47039C8dA53C0be2f err="abi: unmarshalling empty output"
2|node01  | ERROR[04-14|09:25:17] Fail when get all trading pairs          error="abi: unmarshalling empty output"
2|node01  | ERROR[04-14|09:25:17] Fail when process lending liquidation data  error="abi: unmarshalling empty output"
2|node01  | DEBUG[04-14|09:25:18] Invalid stats history request            msg=false
2|node01  | DEBUG[04-14|09:25:18] Recalculated downloader QoS values       rtt=2s           confidence=1.000 ttl=4.000004s
2|node01  | DEBUG[04-14|09:25:18] Order pool status report                 executable=0   queued=0
2|node01  | DEBUG[04-14|09:25:19] Revalidated node                         b=14 id=65051ffdd5da09d8
2|node01  | DEBUG[04-14|09:25:20] Recalculated downloader QoS values       rtt=2s           confidence=1.000 ttl=4.000004s
2|node01  | DEBUG[04-14|09:25:22] Recalculated downloader QoS values       rtt=2s           confidence=1.000 ttl=4.000004s
2|node01  | DEBUG[04-14|09:25:24] Revalidated node                         b=16 id=7d8ffe6d28f738d8

```

### PR DESCRIPTION
**Pull request by thanhnguyennguyen**
**Changes in this PR:**
- Validate token decimal fin TomoXListing tx
- Validate token applying to TomoZ/TomoX not only in `tx_pool` but also in `miner/worker`, `ApplyTransaction`
- Fix chain stop issue in tomoxlending if fail to get token decimals

This fixes #235

### CODE REVIEW

File: tomoxlending/tomoxlending.go [hunk 1]
```diff
@@ -870,7 +870,8 @@ func (l *Lending) ProcessLiquidationData(header *types.Header, chain consensus.C
 		_, collateralPrice, err := l.GetCollateralPrices(header, chain, statedb, tradingState, lendingPair.CollateralToken, lendingPair.LendingToken)
```
Comment (thanhnguyennguyen) [inline tomoxlending/tomoxlending.go:870]: if there is any other failure of `GetCollateralPrices` `GetTokenDecimal` happening in matching engine (tomox/order_processor.go tomoxlending/order_processor.go), order will be added to `reject` and everything keeps working 
All we need to fix is: ignore the token pair in liquidation job below
https://github.com/tomochain/tomochain/pull/236/files#diff-9525e4c0a957e0a3e9764896c772467dR874

**Review by thanhson1085 — APPROVED** [review]

File: core/state_processor.go [hunk 2]
```diff
@@ -93,6 +93,20 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 				return nil, nil, 0, fmt.Errorf("Block contains transaction with receiver in black-list: %v", tx.To().Hex())
 			}
 		}
+		// validate minFee slot for TomoZ
+		if tx.IsTomoZApplyTransaction() {
+			copyState := statedb.Copy()
+			if err := ValidateTomoZApplyTransaction(p.bc, copyState, common.BytesToAddress(tx.Data()[4:])); err != nil {
```
Comment (thanhson1085) [inline core/state_processor.go:99]: seems need to check hardfork for this validation

Comment (thanhnguyennguyen) [reply core/state_processor.go:99]: should we use this one ?https://github.com/tomochain/tomochain/blob/master/common/constants.go#L34

Comment (thanhnguyennguyen) [reply core/state_processor.go:99]: updated !
https://github.com/tomochain/tomochain/pull/236/commits/ceed05816a6058da33d1450bd0af05e03667f5e5

**Pull request merged**